Event Planning and Collaboration Tool/src/main/java/bg/sofia/uni/fmi/web/project/service/ContractEventVendorFacadeService.java

```diff
@@ -0,0 +1,75 @@
+package bg.sofia.uni.fmi.web.project.service;
+
+import bg.sofia.uni.fmi.web.project.model.Contract;
+import bg.sofia.uni.fmi.web.project.model.Event;
+import bg.sofia.uni.fmi.web.project.model.Vendor;
+import bg.sofia.uni.fmi.web.project.validation.ConflictException;
+import bg.sofia.uni.fmi.web.project.validation.ResourceNotFoundException;
+import jakarta.transaction.Transactional;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+@AllArgsConstructor
+public class ContractEventVendorFacadeService {
+    private final ContractService contractService;
+    private final EventService eventService;
+    private final VendorService vendorService;
+
+    @Transactional
+    public long addContract(@NotNull(message = "The given vendor cannot be null!")
```

Comment:
Maybe the checks concerned with validation won’t work without @Validated annotation. The same comment is for some other services.

Reply:
Fixed